utils/training.py

```diff
@@ -133,6 +136,125 @@ def train_shakespeare_transformer(model, train_loader, eval_loader, optimizer, n
     plt.show()
 
 
+def train_nanogpt(model, device, train_data_loader, val_data_loader, max_iters, batch_size):
```

Comment:
Where does the chunking and context window splitting happen in the training function?